main.py

```diff
@@ -0,0 +1,1605 @@
+    
+    '''
+    The for-loop below used to work without issues, but now it's giving an error depending on
+    which message we receive. 
+    Error msg: "Caught exception in on_message: 'list' object has no attribute 'items'"
+    '''
+
+    for key, value in unpacked_json.items():    #iterate for single key-value pair
+        if isinstance(value, dict):
+            for key2, value2 in value.items():  #iterate for a value with value
+                print(key2, ':', value2)
+                curr_line = key2 + ':' + str(value2)
+                message_array.append(curr_line)  #store into array
+                '''Unable to pick "lte" apart or get rid of brackets for "networkInfo"
+                if isinstance(value2, dict):
+                    for key3, value3 in value2.items():  #iterate for a value with a value with value
+                        print(key3, ':', value3)       
+                '''  
+        else:
+            print("{0} : {1}".format(key, value)) 
+            curr_line = key + ': ' + str(value)
+            message_array.append(curr_line)
+    print('\n')
+
+    #pass to function to put into message tab
+    print('message array:', message_array)
+    output_messages(message_array, curr_msg_topic)
+
+def insert_treeview_topic():
+    global curr_msg_topic
+
+    tab3_topic = [] #list for topics in Treeview in Tab3
+    
+    if topic not in tab3_topic:
+        if topic == None:  #for initial subscriptions in the beginning of program
+            short_topic = '...' + address[-10:]  #only show the last 10 characters of a topic
+            curr_msg_topic = address
+            tab3_tree.insert("", END, iid=curr_msg_topic, text=short_topic)
+            tab3_topic.append(address)
+            
+            print('1:', tab3_topic)
+            print('address:', address)
+        else:
+            short_topic = topics.shorten_topic(topic)
+            short_topic = '...' + topic[-10:]
+            tab3_tree.insert("", END, iid=curr_msg_topic, text=short_topic)     #parent
+            tab3_topic.append(topic)
+            
+            print('2:', tab3_topic)
+            print('topic:', topic)
+
+def output_messages(message_array, curr_msg_topic):
+    global j 
+
+    #insert messages in Treeview in its appropriate topic section
+    curr_msg_topic = str(curr_msg_topic)
+    for i in message_array:
+        tab3_tree.insert(parent=curr_msg_topic, index=j, values=i)    #child, insert from top
+        j = j + 1
+
+    '''
+    i = tab3_tree.insert("", END, text="Topic1")
+    #this could be the data topic
+    tab3_tree.insert(parent=i, index=0, iid=0, values=('SubTopic1', 5, 'Time1'), tags='odd')
+    #this could be a data value
+    tab3_tree.insert(parent=i, index=1, iid=1, values=('SubTopic2', 10, 'Time2'), tags='even')
+    j = tab3_tree.insert("", END, text="Topic2")
+    tab3_tree.insert(parent=j, index=2, iid=2, values=('SubTopic3', 3, 'Time3'), tags='odd')
+    '''
+
+'''Tab 2 Stuff'''
```

Comment:
Is there a more descriptive name you can use besides "Tab 2"?  Ditto below for "Tab 1"?

Reply:
👀 
this comment is marked as resolved.
what was the resolution?